src/main/java/com/minecolonies/colony/Schematics.java

```diff
@@ -89,70 +140,77 @@ private static void loadStyleMaps(Path basePath) throws IOException
             {
                 Path path = it.next();
 
-                String style = path.getParent().getFileName().toString();
+                String style = getDirectoryFromPathAfterADirectory(path, SCHEMATIC_ROOT);
 
                 //Don't treat generic schematics as decorations or huts - ex: supply ship
-                if (NULL_STYLE.equals(style))
+                if (SCHEMATIC_ROOT.equals(style) || style.startsWith("_"))
                 {
                     continue;
                 }
 
+                String category = getDirectoryFromPathAfterADirectory(path, style);
+
                 if (path.toString().endsWith(SCHEMATIC_EXTENSION))
                 {
                     String filename = path.getFileName().toString().split("\\.schematic")[0];
                     String hut = filename.split("\\d+")[0];
 
-                    if (isSchematicHut(hut))
+                    StructureInfo info = null;
+
+                    if (!structureMap.containsKey(hut))
                     {
-                        addHutStyle(hut, style);
-                        incrementHutMaxLevel(hut);
+                        Boolean isHut = isSchematicHut(hut);
+                        if (!useNewDirectoryStructure)
+                        {
+                            if (isHut)
+                            {
+                                category = "hut";
+                            }
+                            else
+                            {
+                                category = "decoration";
+                            }
+                        }
+
+                        info = new StructureInfo(hut, isHut, category);
+                        structureMap.put(hut, info);
                     }
                     else
                     {
-                        addDecorationStyle(filename, style);
+                        info = structureMap.get(hut);
                     }
+                    info.addLevel(style, extractInt(filename), path);
                 }
             }
         }
     }
 
-    private static boolean isSchematicHut(String name)
+    private static int extractInt(String fileName)
     {
-        return Block.getBlockFromName(Constants.MOD_ID + ":blockHut" + name) != null;
-    }
-
-    private static void addHutStyle(String hut, String style)
-    {
-        if (!hutStyleMap.containsKey(hut))
-        {
-            hutStyleMap.put(hut, new ArrayList<>());
-        }
-
-        if (!hutStyleMap.get(hut).contains(style))
+        int result = 1;
+        Matcher matcher = Pattern.compile("\\d+").matcher(fileName);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'matcher' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)